### PR TITLE
don't check 0 elements for cat symbolic diff

### PR DESCRIPTION
**Pull request by ngimel**
Summary: Fixes symbolic script grad formula for cat to correctly handle empty tensors

Test Plan: Existing tests

Differential Revision: D31208364

### CODE REVIEW

File: torch/csrc/jit/runtime/symbolic_script.cpp [hunk 1]
```diff
@@ -612,7 +612,7 @@ const std::vector<std::string> functions = {
             size = len(tensors)
             split_sizes = [0] * size
             for i in range(size):
-                if tensors[i].numel() > 0:
+                if tensors[i].numel() > 0 or len(tensors[i].size()) != 1:
```
Comment (eellison) [inline torch/csrc/jit/runtime/symbolic_script.cpp:615]: i guess this is the equivalent logic : https://codebrowser.bddppq.com/pytorch/pytorch/torch/csrc/autograd/FunctionsManual.cpp.html#602

expressed in python as tensors[i].size() != [0]. should we do that here as well ?

Comment (eellison) [inline torch/csrc/jit/runtime/symbolic_script.cpp:615]: nit: `len(tensors[i].size()) != 1` -> `tensors[i].dim() != 1`

**Review by eellison — APPROVED** [review]
lgtm!! thanks!!

File: torch/csrc/jit/runtime/symbolic_script.cpp (hunk 1 above)
Comment (ngimel) [reply torch/csrc/jit/runtime/symbolic_script.cpp:615]: heh, I tried `.ndim` and `.ndimension()`, and neither compiled with script, I guess `.dim()`'s the ticket. But you are right above, these 2 conditions are equivalent to `tensors[i].size() != [0]`

Comment (eellison) [reply torch/csrc/jit/runtime/symbolic_script.cpp:615]: `.size() != [0]` should work, .size() returns a list of ints in TS..

Comment (ngimel) [reply torch/csrc/jit/runtime/symbolic_script.cpp:615]: Ah interesting, it doesn't work in eager
```
In [8]: a.size() == [0]
Out[8]: False

In [9]: a.size() == torch.Size([0])
Out[9]: True
```

**Pull request closed without merging**